warehouse/query-core/src/main/java/datawave/query/jexl/nodes/QueryPropertyMarker.java

```diff
@@ -287,12 +332,57 @@ public boolean isIvarator() {
         }
         
         /**
-         * Return the source node for the query property if this instance is a marker, or null otherwise.
-         *
-         * @return the source node, possibly null
+         * Return the list of all source nodes found if this instance is a marker, or an empty list if no marker was found.
+         * 
+         * @return the list of sources
+         */
+        public List<JexlNode> getSources() {
+            return sources;
+        }
+        
+        /**
+         * Return the total number of sources.
+         * 
+         * @return the number of sources
+         */
+        public int totalSources() {
+            return sources.size();
+        }
+        
+        /**
+         * Return whether there is more than one source.
+         * 
+         * @return true if there is more than one source or false otherwise
+         */
+        public boolean hasMutipleSources() {
+            return totalSources() > 1;
+        }
+        
+        /**
+         * Return a singular source node. If only one source node was found, that specific source will be returned. If multiple source nodes were found, a new
+         * unwrapped {@link ASTAndNode} will be returned with the source nodes as its children. NOTE: original parentage is preserved; the new
+         * {@link ASTAndNode} will not be set as the parent for the source nodes, and thus a node with an invalid structure will be returned if there are
+         * multiple sources.
+         * 
+         * @return a singular source node if this instance is a marker, or null otherwise
          */
         public JexlNode getSource() {
-            return source;
+            int totalSources = sources.size();
+            // If there is only one source node, return that node.
+            if (totalSources == 1) {
+                return sources.get(0);
+            } else if (totalSources > 1) {
+                // If multiple source nodes are found, return them as children of a new AND node.
+                ASTAndNode andNode = new ASTAndNode(ParserTreeConstants.JJTANDNODE);
+                JexlNodes.ensureCapacity(andNode, totalSources);
+                for (int i = 0; i < totalSources; i++) {
+                    andNode.jjtAddChild(sources.get(i), i);
+                    // The parents of the source nodes are intentionally not updated to the new AND node.
+                }
+                return andNode;
```

Comment:
Not for this merge request, but something to consider in the future. If I have a query like `(A || B)` and both terms end up getting delayed, it might be neat to represent that as
 `( _delayed_ && (A or B))` 
instead of 
`( _delayed_ && A) || ( _delayed_ && B)`

Reply:
I'm happy to work on this if you make a ticket for it.

Reply:
For future consideration, not something to be worked in the short term. https://github.com/NationalSecurityAgency/datawave/issues/1304